components/sync15/src/request.rs

```diff
@@ -314,6 +336,8 @@ where
         let can_batch_record = self.batch_limits.can_add_record(payload_length);
         let can_send_record = self.queued.len() < self.max_request_bytes;
 
+        // If any of the limits have been breached,
```

Comment:
This is great comment!

Reword: `If any of the queue limits have been exceeded by enqueueing the record, remove the record, flush the queue, and re-enqueue the record.`